File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.py
class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        n = len(s)
        
        vowels = ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']
        
        a_count = 0
        b_count = 0
        
        for i in range (0, int(n/2)):
            if s[i] in vowels:
                a_count = a_count + 1
                
        for i in range(int(n/2), n):
            if s[i] in vowels:
                b_count = b_count + 1
                
        return a_count == b_count

File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.cpp
class Solution {
public:
    bool halvesAreAlike(string s)
    {
        int a_freq = 0;
        int b_freq = 0;
        
        int n = s.length();
        
        string a = s.substr(0, n/2);
        string b = s.substr(n/2, n);
        
        // vowels of a
        for(auto x: a)
            if(isVowel(x))
                a_freq++;
        
        // vowels of b
        for(auto x: b)
            if(isVowel(x))
                b_freq++;
        
        return a_freq == b_freq;
    }
    
    bool isVowel(char ch)
    {
        vector<char> vowels = {'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'};
        
        return find(vowels.begin(), vowels.end(), ch) != vowels.end();
    }
};